tests/cpp/common/test_column_matrix.cc

```diff
@@ -9,49 +9,110 @@ namespace xgboost {
 namespace common {
 
 TEST(DenseColumn, Test) {
-  auto dmat = RandomDataGenerator(100, 10, 0.0).GenerateDMatix();
-  GHistIndexMatrix gmat;
-  gmat.Init(dmat.get(), 256);
-  ColumnMatrix column_matrix;
-  column_matrix.Init(gmat, 0.2);
+  size_t max_num_bins[] = {256, 65536, 65537};
+  for (size_t max_num_bin : max_num_bins) {
+    auto dmat = RandomDataGenerator(100, 10, 0.0).GenerateDMatix();
+    GHistIndexMatrix gmat;
+    gmat.Init(dmat.get(), max_num_bin);
+    ColumnMatrix column_matrix;
+    column_matrix.Init(gmat, 0.2);
 
-  for (auto i = 0ull; i < dmat->Info().num_row_; i++) {
-    for (auto j = 0ull; j < dmat->Info().num_col_; j++) {
-        auto col = column_matrix.GetColumn(j);
-        ASSERT_EQ(gmat.index[i * dmat->Info().num_col_ + j],
-                  col.GetGlobalBinIdx(i));
+    for (auto i = 0ull; i < dmat->Info().num_row_; i++) {
+      for (auto j = 0ull; j < dmat->Info().num_col_; j++) {
+          switch (column_matrix.GetTypeSize()) {
+            case sizeof(uint8_t): {
+                auto col = column_matrix.GetColumn<uint8_t>(j);
+                ASSERT_EQ(gmat.index[i * dmat->Info().num_col_ + j],
+                          col.GetGlobalBinIdx(i));
+              }
+              break;
+            case sizeof(uint16_t): {
+                auto col = column_matrix.GetColumn<uint16_t>(j);
+                ASSERT_EQ(gmat.index[i * dmat->Info().num_col_ + j],
+                          col.GetGlobalBinIdx(i));
+              }
+              break;
+            case sizeof(uint32_t): {
+                auto col = column_matrix.GetColumn<uint32_t>(j);
+                ASSERT_EQ(gmat.index[i * dmat->Info().num_col_ + j],
+                          col.GetGlobalBinIdx(i));
+              }
+              break;
+        }
+      }
     }
   }
 }
 
-TEST(SparseColumn, Test) {
-  auto dmat = RandomDataGenerator(100, 1, 0.85).GenerateDMatix();
-  GHistIndexMatrix gmat;
-  gmat.Init(dmat.get(), 256);
-  ColumnMatrix column_matrix;
-  column_matrix.Init(gmat, 0.5);
-  auto col = column_matrix.GetColumn(0);
+template<typename BinIdxType>
+inline void CheckSparseColumn(const Column<BinIdxType>& col, const GHistIndexMatrix& gmat) {
   ASSERT_EQ(col.Size(), gmat.index.size());
   for (auto i = 0ull; i < col.Size(); i++) {
     ASSERT_EQ(gmat.index[gmat.row_ptr[col.GetRowIdx(i)]],
               col.GetGlobalBinIdx(i));
   }
 }
 
-TEST(DenseColumnWithMissing, Test) {
-  auto dmat = RandomDataGenerator(100, 1, 0.5).GenerateDMatix();
-  GHistIndexMatrix gmat;
-  gmat.Init(dmat.get(), 256);
-  ColumnMatrix column_matrix;
-  column_matrix.Init(gmat, 0.2);
-  auto col = column_matrix.GetColumn(0);
+TEST(SparseColumn, Test) {
+  size_t max_num_bins[] = {256, 65536, 65537};
+  for (size_t max_num_bin : max_num_bins) {
+    auto dmat = RandomDataGenerator(100, 1, 0.85).GenerateDMatix();
```

Comment:
Similar comment

Reply:
to be honest it makes sense, as the selection of appropriate type (uint8, uint16, uint32) is done by max_bins value only. I want to have this check very fast :)

Reply:
Probably, but I think we can test this for 500 rows for example, to have more than 255 bins